packages/openapi-generator/src/model.spec.ts

```diff
@@ -0,0 +1,123 @@
+import {
+  collectRefs,
+  hasNotSchema,
+  parseFileNameFromRef,
+  parseTypeNameFromRef
+} from './model';
+
+describe('parseTypeNameFromRef', () => {
+  it('gets the last part of a reference as type name', () => {
+    expect(
+      parseTypeNameFromRef({
+        $ref: '#/components/schemas/typeName'
+      })
+    ).toEqual('TypeName');
+  });
+
+  it('gets the last part of a string as type name', () => {
+    expect(parseTypeNameFromRef('#/components/schemas/typeName')).toEqual(
+      'TypeName'
+    );
+  });
+});
+
+describe('parseFileNameFromRef', () => {
+  it('gets the last part of a reference as type name', () => {
+    expect(
+      parseFileNameFromRef({
+        $ref: '#/components/schemas/typeName'
+      })
+    ).toEqual('type-name');
+  });
+
+  it('gets the last part of a string as type name', () => {
+    expect(parseFileNameFromRef('#/components/schemas/typeName')).toEqual(
+      'type-name'
+    );
+  });
+});
+
+describe('hasNotSchema', () => {
+  it('returns true if there is at least one descendant with a not schema', () => {
+    expect(
+      hasNotSchema({
+        oneOf: [
+          { type: 'string' },
+          {
+            type: 'object',
+            properties: [
+              {
+                name: 'notProperty',
+                required: false,
+                schema: { not: { type: 'number' } }
+              }
+            ]
+          }
+        ]
+      })
+    ).toBe(true);
+  });
+
+  it('returns true if the root schema is not schema', () => {
```

Comment:
Add test if the not is in a referenced schema.

Reply:
Same as above.